### DISCUSSION TITLE
Details about Metrics Tool

### DISCUSSION THREAD

**Discussion by MinaHey** [PlayForDreamDevelopers/support#21] [General]
we got some KPI in logcat. Is there anyone can share the detailed illustration for them? or how we can get to know it? what CPU level 4 means? what GPU level 5? how is CPU and GPU utilization calculated, etc...
_DynamicLog: PerfStats.cpp::statsPerfInfo: FPS=89.96/90, Prd=34.57ms, Early=0.00, Stale=0.00, Lat=-1, Fov=0, CPU/GPU=4/5, Free=2938MB, ATW=0.00ms, APP=4.12ms/5.14ms, GD=0.00, CPU%=0.40, GPU%=0.57_

**Comment by YVRDevelopers** [comment]
Hi there, here's the detailed illustration for the metrics:

1. **GPU Utilization (GPU U)**: Displays the GPU utilization percentage. If this value is maxed 100%, the app is GPU bound. Performance issues due to scheduling may occur if this number is over 90%.

2. **CPU Utilization (CPU U)**: Displays the CPU utilization percentage. However, this metric represents the worst performing core, and since most apps are multithreaded, and the scheduler assigns threads as they are available, the main thread of the app may not be represented in this metric.

3. **CPU Level (CPU L) & GPU Level (GPU L)**: These metrics specify the CPU and GPU clock levels set by the app. While these levels can be set manually, use of dynamic clock throttling, which increases these numbers if the app is not hitting frame rate at a requested level, is recommended.
If an app is not making frame rate, reviewing the clock levels can help quickly indicate if performance is CPU or GPU bound, providing a target area for optimization. For example, if an app with performance issues is at CPU 4 and GPU 2, the app must be CPU bound since there is still available GPU overhead. However, if both levels are 4 and the app has issues, this number is not as useful, and other metrics should be used to find potential areas for optimization, such as stale frames, app GPU time, and CPU/GPU utilization.

**Reply by MinaHey** [reply]
Thanks @YVRDevelopers for elaboration! Regarding CPU utilization, could you please help clarify how to decide the worst performing core as each core of a cluster should have various capacity? 
Besides, it would be helpful for us if you could share more details about other KPIs like "ATW=0.00ms, APP=4.12ms/5.14ms". What 4.12ms and 5.14ms indicate here? We get consistent ATW=0. Is this KPI enabled? Is any required property needs to be set? 
Thank you again for help!

**Comment by YVRDevelopers** [comment]
So ATW isn't available. As for APP=4.12ms/5.14ms, 4.12ms = CPU Time, 5.14ms = GPU Time.
